core/src/main/java/com/datastax/oss/driver/internal/core/metadata/token/CDCTokenFactory.java

```diff
@@ -0,0 +1,84 @@
+package com.datastax.oss.driver.internal.core.metadata.token;
+
+import com.datastax.oss.driver.api.core.metadata.token.Token;
+import com.datastax.oss.driver.api.core.metadata.token.TokenFactory;
+import com.datastax.oss.driver.api.core.metadata.token.TokenRange;
+import com.datastax.oss.driver.shaded.guava.common.base.Preconditions;
+import java.nio.ByteBuffer;
+import net.jcip.annotations.ThreadSafe;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ThreadSafe
+public class CDCTokenFactory implements TokenFactory {
+
+  public static final String PARTITIONER_NAME = "com.scylladb.dht.CDCPartitioner";
+
+  public static final CDCToken MIN_TOKEN = new CDCToken(Long.MIN_VALUE);
+  public static final CDCToken MAX_TOKEN = new CDCToken(Long.MAX_VALUE);
+
+  private static final Logger logger = LoggerFactory.getLogger(CDCTokenFactory.class);
+  private static final int CDC_PARTITION_KEY_LENGTH = 16;
+  private static final long VERSION_MASK = 0xF;
+  private static final int MIN_SUPPORTED_VERSION = 1;
+  private static final int MAX_SUPPORTED_VERSION = 1;
+
+  @Override
+  public String getPartitionerName() {
+    return PARTITIONER_NAME;
+  }
+
+  @Override
+  public Token hash(ByteBuffer partitionKey) {
+    int offset = partitionKey.position();
+    int length = partitionKey.remaining();
+
+    if (length != CDC_PARTITION_KEY_LENGTH) {
+      logger.warn(
+          "CDC partition key has invalid length: expected {} bytes, but got {} bytes",
+          CDC_PARTITION_KEY_LENGTH,
+          length);
+    }
+    if (length < 8) {
+      return MIN_TOKEN;
+    }
+
+    long upperDword = partitionKey.getLong(offset + 0);
```

Comment:
offset + 0?

Reply:
To indicate that `offset` is a starting offset and we will read first long (`upperDword`) and second long (`lowerDword`), explicitly stating the "internal" offsets of those two values.

Reply:
I don't see much value in it but ok.